graql/reasoner/query/DisjunctiveQuery.java

```diff
@@ -0,0 +1,277 @@
+    public ResolvableQuery withSubstitution(ConceptMap sub) {
+        return new DisjunctiveQuery(
+                getClauses().stream().map(q -> q.withSubstitution(sub)).collect(Collectors.toSet()),
+                getBindingVars(),
+                traversalExecutor,
+                context()
+        );
+    }
+
+    @Override
+    public Stream<ConceptMap> traverse(){
+        Stream<Stream<ConceptMap>> answerStreams = clauses.stream().map(clause ->
+                clause.traverse().map(ans -> {
+                    ConceptMap clauseAns = new ConceptMap(ans.map(), new LookupExplanation(), clause.getPattern(ans.map()));
+                    HashMap<Variable, Concept> bindingVarsSub = filterBindingVars(ans.map());
+                    return new ConceptMap(bindingVarsSub, new DisjunctiveExplanation(clauseAns), getPattern(bindingVarsSub));
+                }));
+        LazyMergingStream<ConceptMap> mergedStreams = new LazyMergingStream<>(answerStreams);
+        return mergedStreams.flatStream();
+    }
+
+    @Override
+    ResolvableQuery inferTypes() {
+        return new DisjunctiveQuery(
+                getClauses().stream().map(ResolvableQuery::inferTypes).collect(Collectors.toSet()),
+                getBindingVars(),
+                traversalExecutor,
+                context()
+        );
+    }
+
+    @Override
+    ResolvableQuery constantValuePredicateQuery() {
+        return new DisjunctiveQuery(
+                getClauses().stream().map(ResolvableQuery::constantValuePredicateQuery).collect(Collectors.toSet()),
+                getBindingVars(),
+                traversalExecutor,
+                context()
+        );
+    }
+
+    @Override
+    ResolvableQuery copy() {
+        return new DisjunctiveQuery(
+                getClauses(),
+                getBindingVars(),
+                traversalExecutor,
+                context());
+    }
+
+    @Override
+    public boolean isAtomic() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    boolean isEquivalent(ResolvableQuery q) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String toString(){
+        return getPattern().toString();
+    }
+
+    @Override
+    public ReasonerQuery conjunction(ReasonerQuery q) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void checkValid() {
+        clauses.forEach(ResolvableQuery::checkValid);
+    }
+
+    @Override
+    public Set<Variable> getVarNames() {
+        return bindingVars;
+    }
+
+    @Override
+    public Set<Atomic> getAtoms() {
+        return getClauses().stream().flatMap(c -> c.getAtoms().stream()).collect(Collectors.toSet());
+    }
+
+    @Override
+    public Disjunction<Pattern> getPattern() {
+        return Graql.or(clauses.stream().map(ResolvableQuery::getPattern).collect(Collectors.toSet()));
+    }
+
+    @Override
+    public Conjunction<Pattern> getPattern(Map<Variable, Concept> map) {
```

Comment:
you think this name is sufficient? I would at least call the `map` the `substitution` in line with other reasoner terms?